libraries/AC_Avoidance/AC_Avoid.cpp

```diff
@@ -1246,14 +1221,7 @@ void AC_Avoid::adjust_velocity_proximity(float kP, float accel_cmss, Vector3f &d
                     // i.e. do not adjust velocity.
                     return;
                 }
-                if (limit_distance_cm <= margin_cm) {
-                    // we are within the margin so stop vehicle
-                    safe_vel.zero();
-                } else {
-                    // vehicle inside the given edge, adjust velocity to not violate this edge
-                    limit_velocity_3D(kP, accel_cmss, safe_vel, limit_direction, margin_cm, kP_z, accel_cmss_z, dt);
-                }
-
+                limit_velocity_3D(kP, accel_cmss, safe_vel, limit_direction, margin_cm, kP_z, accel_cmss_z, dt);
```

Comment:
@rishabsingh3003 I think this might be the issue for obstacles. The safe_vel is zero and this overrides the desired speed to zero hence rover stops with no control.
I was able to generate the issue and it disappeared when I removed this line. I tried twice.
I think this might be in other parts as well but I was able to test it with range finder as I did it indoors with no geo fences.
kindly advise.

Reply:
@HefnySco again, I need logs and videos to understand what you are talking about. The safe_vel.zero() is only done with we are inside the margin. That is what "mode stop" is all about, so it is needed.

Reply:
@rishabsingh3003 
This is the [video](https://youtu.be/pYGbhH_edC8?t=76) that shows the issue. Please check time 1:13 and beyond.
This is the [video](https://youtu.be/nAjDFAATC5U) with updated software.
Log files are [here](https://drive.google.com/drive/folders/1IUvwEqxudSTkzvxOM_KsU8SnddWcsj_s?usp=sharing).

"mode stop" stops the vehicle by setting desired speed to zero. so when you use desired speed to allows vechile to obey user movement it fails to move. 
Thanks

Reply:
@rishabsingh3003 kindly advise.